core/src/main/java/de/medavis/lct/core/patcher/AbstractRestClient.java

```diff
@@ -0,0 +1,128 @@
+/*-
+ * #%L
+ * License Compliance Tool - Implementation Core
+ * %%
+ * Copyright (C) 2022 - 2024 medavis GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package de.medavis.lct.core.patcher;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import de.medavis.lct.core.Json5MapperFactory;
+
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.List;
+
+public abstract class AbstractRestClient {
```

Comment:
This class is a generic utility that is used beyond the scope of BOM patching. It should be in a top-level package on its own.